spec/ttt/core_spec.clj

```diff
@@ -1,59 +1,70 @@
+          (with-redefs [db/update-stats-when-winner (stub :db/update-stats-when-winner {:return 1})
+                        db/retrieve-all-player-stats (stub :db/retrieve-all-player-stats {:return players-stats})]
+          (should-invoke ui/print-win-message {:with ["Jon"]} (game/play initial-state)))))
+
+    (it "makes a move when the game is not over"
+        (with-redefs [player/get-move (stub :player/get-move {:return 4})
+                      game/set-game-state (stub :game/set-game-state {:return state-after-one-move})
+                      db/update-stats-when-winner (stub :db/update-stats-when-winner {:return 1})
+                      db/retrieve-all-player-stats (stub :db/retrieve-all-player-stats {:return players-stats})
+                      rules/win? (let [results (atom [false true])]
+                                         (fn [_] (ffirst (swap-vals! results rest))))
+                      rules/tie? (let [results (atom [false true])]
+                                         (fn [_] (ffirst (swap-vals! results rest))))]
+          (should-invoke board/make-move {:with [[0 1 2 3 4 5 6 7 8] 4 "X"]} (game/play initial-state))))
+
+    (it "prints the new board after a move is made"
+        (with-redefs [player/get-move (stub :player/get-move {:return 1})
+                      board/make-move (stub :board/make-move {:return [0 "X" 2 3 4 5 6 7 8]})
+                      db/update-stats-when-winner (stub :db/update-stats-when-winner {:return 1})
+                      db/retrieve-all-player-stats (stub :db/retrieve-all-player-stats {:return players-stats})
+                      rules/win? (let [results (atom [false true])]
+                                   (fn [_] (ffirst (swap-vals! results rest))))
+                      rules/tie? (let [results (atom [false false])]
+                                   (fn [_] (ffirst (swap-vals! results rest))))]
+          (should-contain boards-printed-after-one-move (with-out-str (game/play initial-state)))))
+
+    (it "prints new boards after each player makes a move"
+        (with-redefs [player/get-move (let [results (atom [4 8])]
+                                        (fn [] (ffirst (swap-vals! results rest))))
+                      board/make-move (let [results (atom [[0 1 2 3 "X" 5 6 7 8] [0 1 2 3 "X" 5 6 7 "O"]])]
+                                        (fn [_a _b _c] (ffirst (swap-vals! results rest))))
+                      db/update-stats-when-winner (stub :db/update-stats-when-winner {:return 1})
+                      db/retrieve-all-player-stats (stub :db/retrieve-all-player-stats {:return players-stats})
+                      rules/win? (let [results (atom [false false true])]
+                                    (fn [_] (ffirst (swap-vals! results rest))))
+                        rules/tie? (let [results (atom [false false true])]
+                                    (fn [_] (ffirst (swap-vals! results rest))))]
```

Comment:
since we're doing this quite a few times, could we try and extract this out into a function for reuse?